### PR TITLE
Feature/conditional granger

### PR DESCRIPTION
**Pull request by Qkley**
This is the PR for conditional GC. We would like to have it reviewed. Please let us know if the number of tests for the new function is sufficient. Also, it would be good to check the documentation for compliance with the Elephant standard.

### CODE REVIEW

File: elephant/causality/granger.py [hunk 1]
```diff
@@ -526,3 +526,73 @@ def pairwise_granger(signals, max_order, information_criterion='aic'):
...
+
+    # Round conditional GC according to following scheme:
+    #     Note that standard error scales as 1/sqrt(sample_size)
+    #     Calculate  significant figures according to standard error
+    length = np.size(signal_x)
+    asymptotic_std_error = 1/np.sqrt(length)
+    est_sig_figures = int((-1)*np.around(np.log10(asymptotic_std_error)))
+
+    conditional_causality_xy_z_round = np.around(conditional_causality_xy_z,
+                                                 est_sig_figures)
```
Comment (dizcza) [inline elephant/causality/granger.py:596]: why do you round the resulting value? does it make sense other than more human-readable?

File: elephant/causality/granger.py [hunk 2]
```diff
@@ -526,3 +526,73 @@ def pairwise_granger(signals, max_order, information_criterion='aic'):
...
+
+    # transpose (N,3) -> (3,N) for mathematical convenience
+    signals = signals.T
+
+    # signal_x, signal_y and signal_z are (1, N) arrays
+    signal_x, signal_y, signal_z = np.expand_dims(signals, axis=1)
+
+    signals_xz = np.vstack([signal_x, signal_z])
+
+    coeffs_xz, cov_xz, p_1 = _optimal_vector_arm(signals_xz, 2, max_order,
```
Comment (dizcza) [inline elephant/causality/granger.py:581]: use keywords
```
dimension=2,
max_order=max_order,.
...
```
same for dimension 3

File: elephant/causality/granger.py [hunk 3]
```diff
@@ -45,6 +45,7 @@
     :toctree: toctree/causality/
 
     pairwise_granger
+    conditional_granger
```
Comment (dizcza) [inline elephant/causality/granger.py:48]: add conditional_granger in `__all__` as well

**Review by dizcza — COMMENTED** [review]
The implementation and the docs look good, though maybe Cristiano has a better eye for it.

One thing to add:
Currently, you have provided only one test that checks for zero causality of the ground truth data. Add at least 1 test that checks for non-zero causality (simulate the data).

File: elephant/causality/granger.py (hunk 1 above)
Comment (Qkley) [reply elephant/causality/granger.py:596]: We use rounding to avoid tiny negative causality results, when the actual output should be zero. This is also done in the pairwise causality method as well (https://github.com/NeuralEnsemble/elephant/pull/332/commits/e577ce0b891e010d3a3b59ca4a2de775cc5cb493).

Comment (dizcza) [reply elephant/causality/granger.py:596]: The rounding in pairwise granger didn't catch my eye.

All right, it'd be nice if you put your answer into a code comment describing "We use rounding to avoid tiny negative causality results, when the actual output should be zero". Such a comment will help future developers to read the code. This is optional.

File: elephant/causality/granger.py (hunk 2 above)
Comment (Qkley) [reply elephant/causality/granger.py]: Addressed in commit https://github.com/NeuralEnsemble/elephant/pull/359/commits/9a8d6272b94511ece1fa2a4ebda7e233fb5d0fa5

File: elephant/causality/granger.py (hunk 3 above)
Comment (Qkley) [reply elephant/causality/granger.py:48]: Addressed in commit https://github.com/NeuralEnsemble/elephant/pull/359/commits/bb4d8fe77adcbefa1bd4944c87376c4c1bedcbb5

**Comment by Qkley** [comment]
The comments have been addressed. In particular, I have extended the conditional causality test case to include non-zero conditional causality unit test.
